### PR TITLE
Track updates of globals that are exported as default

### LINKED ISSUE #3534: Exports moved unexpectedly with preserveModules
- Rollup Version: 2.7.6
- Operating System (or Browser): OSX
- Node Version (if applicable): 12.16.1
- Link to reproduction (IMPORTANT, read below): https://repl.it/repls/SpanishImmaculatePrediction

### Input

**index.js**

```js
import bar, { foo } from "./lib";
console.log(bar, foo);
```

**lib.js**

```js
export default SOME_VARIABLE;
export const foo = SOME_OTHER_VARIABLE;
```

### Expected Behavior

`SOME_VARIABLE` and `SOME_OTHER_VARIABLE` remain within `lib.js`.

### Actual Behavior

**index.js**

```js
import { foo } from './lib.js';

console.log(SOME_VARIABLE, foo);
```

This change happened after updating from Rollup 2.2.0. In 2.2.0 it behaves as expected, but I'm not sure exactly which version introduced the behaviour change.

This transform feels like it violates `preserveModules`.

### PR DESCRIPTION
**Pull request by lukastaegert**
This PR contains:
- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?
- [x] yes (*bugfixes and features will not be merged without tests*)
- [ ] no

Breaking Changes?
- [ ] yes (*breaking changes will not be merged unless absolutely necessary*)
- [x] no

List any relevant issue numbers:
Resolves #3534

### Description
When a global variable is assigned to a default export, Rollup does not track updates of the global variable even though it should, creating an unintended live-binding to the global variable where you would actually want a snapshot:
https://rollupjs.org/repl/?version=2.9.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMCU3QmRlZmF1bHQlMjBhcyUyMG9yaWdpbmFsJTJDJTIwdXBkYXRlZCU3RCUyMGZyb20lMjAnLiUyRmxpYiclM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJsaWIuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwZGVmYXVsdCUyMG15R2xvYmFsJTNCJTVDbm15R2xvYmFsJTIwJTNEJTIwMyUzQiU1Q25leHBvcnQlMjBjb25zdCUyMHVwZGF0ZWQlMjAlM0QlMjBteUdsb2JhbCUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJjanMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

or even creating invalid code for ESM as you cannot directly reexport a global variable binding from a module. This is fixed by treating all global variables as having been (potentially) reassigned, which is true semantically.

Even though this is not the original issue of #3534, that issue will be solved here as well.

### CODE REVIEW

File: test/chunking-form/samples/export-default-global/_expected/es/main.js [hunk 1]
```diff
@@ -1,3 +0,0 @@
-console.log(global);
-
-export default global;
```
Comment (guybedford) [inline test/chunking-form/samples/export-default-global/_expected/es/main.js:3]: This test case very much should work - I don't think this should be removed.

**Comment by guybedford** [comment]
Also see previous bug - https://github.com/rollup/rollup/issues/2000.

File: test/chunking-form/samples/export-default-global/_expected/es/main.js (hunk 1 above)
Comment (lukastaegert) [reply test/chunking-form/samples/export-default-global/_expected/es/main.js:3]: It was not removed, it was just moved elsewhere. But I actually moved it back now.

File: test/function/samples/default-exported-global/_config.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const assert = require('assert');
+
+module.exports = {
+	description: 'Tracks updates of default exported globals',
+	context: {
+		myGlobal: 42
+	},
+	exports(exports) {
+		assert.deepStrictEqual(exports, { original: 42, updated: 1 });
```
Comment (guybedford) [inline test/function/samples/default-exported-global/_config.js:9]: Ok, this looks correct!

**Comment by lukastaegert** [comment]
> Note that in ES modules, global = value is not a supported global assignment (strict mode!)

Ah I see, but I think I can rework the test case so that we update the global variable in a legal way.

File: test/chunking-form/samples/export-default-global/_expected/es/main.js (hunk 1 above)
Comment (guybedford) [reply test/chunking-form/samples/export-default-global/_expected/es/main.js:3]: I see, looks good.

**Comment by lukastaegert** [comment]
Unfortunately #2000 was not really resolved, at least the examples are still not working: https://rollupjs.org/repl/?circleci=10966&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMnZhciUyMHglMjAlM0QlMjAxMCUzQiU1Q24lNUNuZXhwb3J0JTIwZnVuY3Rpb24lMjB1cGRhdGUlMjAoKSUyMCU3QiU1Q24lNUN0eCUyMCUzRCUyMDExJTNCJTVDbiU3RCU1Q24lNUNuZXhwb3J0JTIwJTdCJTIweCUyMGFzJTIwZGVmYXVsdCUyMCU3RCUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

**Comment by guybedford** [comment]
I see, yes I misread that - I did think I had posted an issue for this snapshotting exactly, but apparently not.

**Pull request merged**